packages/core/src/conflict-resolution.ts

```diff
@@ -389,4 +377,19 @@ export class ConflictResolution {
     const handler = this.handlers.get(initialState.type)
     return this.applyLogToState(handler, resetLog)
   }
+
+  /**
+   * Return the log entry with its commit and JWS envelope (if applicable) populated if they're not already present.
+   */
+  private async expandLogEntry(logEntry: LogEntry): Promise<LogEntry> {
+    if (!logEntry.commit) {
+      const commit = await this.dispatcher.retrieveCommit(logEntry.cid)
+      logEntry.commit = commit
+      if (StreamUtils.isSignedCommit(commit)) {
+        logEntry.commit = await this.dispatcher.retrieveCommit(commit.link)
+        logEntry.envelope = commit
```

Comment:
This seems backwards? Shouldn't the JWS be the envelope?

Reply:
Ah, sorry I misread!

Reply:
Hmm, isn't that what this code is doing? The naming here has definitely been very confusing for me to follow
 `commit` could be a `SignedCommit` and, if so, it is stored in `envelope`.

Reply:
You are right, I misread the code.